napari/_qt/containers/qt_axis_model.py

```diff
@@ -0,0 +1,178 @@
+from typing import Any, Iterable
+
+import qtpy
+from qtpy.QtCore import QModelIndex, Qt
+
+from napari._qt.containers.qt_list_model import QtListModel
+from napari.components import Dims
+from napari.utils.events import SelectableEventedList
+
+# add PyQt6 ItemFlags compatibility
+if qtpy.API_NAME == 'PyQt6':
+    Qt.ItemFlags = Qt.ItemFlag
+
+
+class AxisModel:
+    """View of an axis within a dims model.
+
+    The model keeps track of axis names and allows read / write
+    acces on the cossesponding rollable state of a Dims object.
+
+    Parameters
+    ----------
+    dims : napari.components.dims.Dims
+        Parent Dims object.
+    axis : int
+        Axis index.
+
+    Attributes
+    ----------
+    dims : napari.components.dims.Dims
+        Dimensions object modeling slicing and displaying.
+    axis : int
+        Axis index.
+    """
+
+    def __init__(self, dims: Dims, axis: int) -> None:
+        self.dims = dims
+        self.axis = axis
+
+    def __hash__(self) -> int:
+        return id(self)
+
+    def __str__(self) -> str:
+        return repr(self)
+
+    def __repr__(self) -> str:
+        return self.dims.axis_labels[self.axis]
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, int):
+            return self.axis == other
+        if isinstance(other, str):
+            return repr(self) == other
+        if isinstance(other, AxisModel):
+            return (self.dims is other.dims) and (self.axis == other.axis)
+        return NotImplemented
+
+    @property
+    def rollable(self) -> bool:
+        """Get or set the 'rollable' state.
+
+        Parameters
+        ----------
+        value : bool
+            True if axis should be rollable.
+
+        Returns
+        -------
+        bool
+            True if axis is rollable.
+        """
```

Comment:
```suggestion
        """
        If the axis should be rollable.
        """
```
as user use it with assignment, not calling function